ui/src/app/subnets/views/VLANDetails/VLANSummary/VLANSummary.tsx

```diff
@@ -1,9 +1,135 @@
-import { Strip } from "@canonical/react-components";
+import {
+  Col,
+  Icon,
+  Row,
+  Spinner,
+  Strip,
+  Tooltip,
+} from "@canonical/react-components";
+import { useSelector } from "react-redux";
+import { Link } from "react-router-dom";
 
-const VLANSummary = (): JSX.Element => (
-  <Strip shallow>
-    <h2 className="p-heading--4">VLAN summary</h2>
-  </Strip>
-);
+import ControllerLink from "app/base/components/ControllerLink";
+import Definition from "app/base/components/Definition";
+import controllerSelectors from "app/store/controller/selectors";
+import fabricSelectors from "app/store/fabric/selectors";
+import { getFabricDisplay } from "app/store/fabric/utils";
+import type { RootState } from "app/store/root/types";
+import spaceSelectors from "app/store/space/selectors";
+import { getSpaceDisplay } from "app/store/space/utils";
+import vlanSelectors from "app/store/vlan/selectors";
+import type { VLAN, VLANMeta } from "app/store/vlan/types";
+import subnetsURLs from "app/subnets/urls";
+import { breakLines } from "app/utils";
+
+type Props = {
+  id: VLAN[VLANMeta.PK] | null;
+};
+
+const getRackIDs = (vlan: VLAN | null) => {
+  const rackIDs = [];
+  if (vlan) {
+    if (vlan.primary_rack) {
+      rackIDs.push(vlan.primary_rack);
+    }
+    if (vlan.secondary_rack) {
+      rackIDs.push(vlan.secondary_rack);
+    }
+  }
+  return rackIDs;
+};
+
+const VLANSummary = ({ id }: Props): JSX.Element | null => {
+  const vlan = useSelector((state: RootState) =>
+    vlanSelectors.getById(state, id)
+  );
+  const controllers = useSelector((state: RootState) =>
+    controllerSelectors.getByIDs(state, getRackIDs(vlan))
+  );
+  const controllersLoading = useSelector(controllerSelectors.loading);
+  const fabric = useSelector((state: RootState) =>
+    fabricSelectors.getById(state, vlan?.fabric)
+  );
+  const fabricsLoading = useSelector(fabricSelectors.loading);
+  const space = useSelector((state: RootState) =>
+    spaceSelectors.getById(state, vlan?.space)
+  );
+  const spacesLoading = useSelector(spaceSelectors.loading);
+
+  if (!vlan) {
+    return null;
+  }
+
+  const fabricDisplay = getFabricDisplay(fabric);
+  const spaceDisplay = getSpaceDisplay(space);
+  return (
+    <Strip shallow>
+      <h2 className="p-heading--4">VLAN summary</h2>
```

Comment:
Would you mind using a semantic element here? Using a `section` element that's described by the heading will help accessibility and allow being more specific in tests:

```jsx
  const id = nanoid();
  return (
    <Strip shallow element="section" aria-labelledby={id}>
      <h2 id={id} className="p-heading--4">VLAN summary</h2>
```

in tests:
```jsx
  const vlanSummary = screen.getByRole("region", { name: "VLAN summary" });

  expect(within(vlanSummary).getByRole("link", { name: space.name })).toHaveAttribute(
    "href",
    subnetsURLs.space.index({ id: space.id })
  );
```

Reply:
SpaceSummary for reference:

- https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/subnets/views/SpaceDetails/SpaceSummary/SpaceSummary.tsx
- https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/subnets/views/SpaceDetails/SpaceSummary/SpaceSummary.test.tsx

Reply:
Sure!

Reply:
It might be best to stick the `nanoid()` call in a `useRef` so the id isn't changed on each render.